app/src/main/java/org/mozilla/focus/tabs/MergeTabsMiddleware.kt

```diff
@@ -31,6 +33,25 @@ class MergeTabsMiddleware(
         next: (BrowserAction) -> Unit,
         action: BrowserAction
     ) {
+        if (!hasTabs && action is ConsumeWindowRequestAction) {
+            val isRequestToCloseWindow =
+                context.state.privateTabs.first().content.windowRequest?.type == WindowRequest.Type.CLOSE
```

Comment:
Tested on https://jsfiddle.net/da5z3egp/ and saw that while on desktop the window will be closed there is no such callback coming from GV for us.
Not sure if/how this request would get to use but the app should be ready to handle it.
In the above video I used a delayed `this.context.components.tabsUseCases.removeAllTabs()` to see how the app would act in the event such a `CLOSE` request would come to the app after a `OPEN` request was acted upon.

Reply:
Here's an alternative that should work fine: https://jsfiddle.net/8dp4huag/1/

We should file a follow-up to investigate why the close call doesn't work in that `setTimeout`, but that's not a blocker for this.